Realm/RLMRealmConfiguration.mm

```diff
@@ -152,15 +151,21 @@ - (NSString *)description {
 }
 
 - (void)setInMemoryIdentifier:(NSString *)inMemoryIdentifier {
-    if ((_inMemoryIdentifier = inMemoryIdentifier)) {
-        _path = nil;
+    if (inMemoryIdentifier.length == 0) {
+        @throw RLMException(@"In-memory identifier must not be empty");
     }
+
+    _inMemoryIdentifier = inMemoryIdentifier;
```

Comment:
Should we be copying `inMemoryIdentifier` in case we're passed a mutable string?